components/tabs/tabs.js

```diff
@@ -1128,10 +1128,11 @@ class Tabs extends LocalizeCoreElement(ArrowKeysMixin(SkeletonMixin(LitElement))
 	}
 
 	async #updateSelectedTab(selectedTab) {
-		selectedTab.tabIndex = 0;
-
 		await this.updateComplete;
 
+		selectedTab.selected = true;
+		selectedTab.tabIndex = 0;
```

Comment:
I noticed a timing issue with the "with slot" demo with the new structure where the newly added tab was triggering the `d2l-tab-selected` event, but in slotchange the new tab wasn't selected yet so it marked the "All" tab as the selected one, then they both ran this function and both ended up de-selected.

This happened if I clicked "Remove" and then "Add Selected" in the "with slot" demo.